xarray/core/coordinates.py

```diff
@@ -246,8 +246,12 @@ def _update_coords(
     def __delitem__(self, key: Hashable) -> None:
         if key in self:
             del self._data[key]
+            if key in self._data.indexes:
+                indexes = dict(self._data.indexes)
+                indexes.pop(key)
+                self._data._indexes = indexes
```

Comment:
`Dataset.__delitem__` already removes indexes, so I believe this is redundant.

Reply:
:+1: